docs/04-Scripts.md

```diff
@@ -1,4 +1,159 @@
+
+=head1 BUGS
+
+None reported (or list of bugs)
+
+=head1 LICENSING
+
+License: GPLv3
+
+=head1 AUTHORS
+
+LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative 
+Neuroscience 
+
+=cut
+
+</pre>
+
+## 4.2 - Pipeline flow
+
+A very brief illustration of the key and expected outcomes at different steps in 
+the execution of the pipeline are shown in 
+![pipeline_flow](images/pipeline_schematic.png). This figure highlights the few 
+key scripts in the pipeline execution, the order in which they call each other, 
+and the interactions with the diverse libraries and utility classes. It 
+therefore serves as a tool to help project developers understand the pipeline
+quickly, and develop independently project customizations and debugging skills.
+
+## 4.3 - Common insertion scripts re-run
+
+The flow in the diagram above corresponds to a new upload. Occasionally, 
+however, projects might need to modify (partially or in full) the outcome of an 
+inserted scan. The next section highlights the "how-to" of the most commonly 
+encountered cases. 
+
+#### Bypassing protocol violation checks
+
+For cases when a scan has triggered a protocol violation, the MINC volume can be
+  **force-loaded** into LORIS by running:
+```
+uploadNeuroDB/minc_insertion.pl -acquisition_protocol t2w -bypass_extra_file_checks -create_minc_pics -profile prod -globLocation -force  -tarchivePath /data/project/dataTransfer/library/2009/DCM_2009-09-25_project_20110214_185904581.tar -mincPath /data/project/data/trashbin/TarLoad-3-34-pVzGC5/xxx0067_703739_v12_20090925_222403_18e1_mri.mnc
+```
+
+Note carefully the following arguments:
+
+- *acquisition_protocol*: must be a known scan type according to the
+    `mri_scan_type` table
+- *tarchive_Path*: the DICOM tarball
+- *mincPath*: note this file may have been placed in the `trashbin` directory
+
+See also: [MRI-PR#141](https://github.com/aces/Loris-MRI/pull/141) for more
+  examples.
+
+#### Rerunning the Imaging pipeline
+
+- If one of the final steps such as the MINC conversion is failing, you may
+    wish to just re-run the tarchiveLoader script.
+    
+> When the need arises to re-load imaging data in LORIS, it is generally not
+   sufficient to just re-run the MINC/NIfTI loading step (`tarchiveLoader` or
+    `batch_uploads_tarchive`). The pipeline steps must be re-run starting
+    with dicomTar.pl (see section 5.4 of
+   [Pipeline triggering options](05-PipelineOptions.md)).
+
+In general, to re-load an imaging dataset through the pipeline from start (from
+  `dicomTar.pl`) -- Ensure entries from the previous attempt to load the dataset
+  have been removed from the following database tables:
+
+- `parameter_file`
+- `tarchive`
+- `mri_acquisition_dates`
+- `files` (best to delete from this table last)
+- `mri_upload`
+- `session` - not recommended - only if necessary, and only if no other data is
+    associated to this session e.g. on the Behavioural side of Loris.
```

Comment:
with the new foreign keys, the order of deletion should probably be:

1) parameter_file
2) mri_acquisition_dates
3) files
4) mri_upload
5) session
6) tarchive
 
for the qc info, should be deleted before 3) files

Reply:
Ok, moved tarchive to the end and added a note about the deletion of the QC having to be done before deleting from `files`